tests/serving/test_serving.py

```diff
@@ -910,3 +911,70 @@ def test_serialize():
     # simulate mlrun/__main__.py
     eval_fn_result = py_eval(str(fn.to_dict()))
     mlrun.utils.helpers.as_dict(eval_fn_result)
+
+
+def generate_field(event):
+    event[event["field_name"]] = event["field_targets"]
+    return event
+
+
+@pytest.mark.parametrize("field_name", ["fieldA", "fieldB"])
+@pytest.mark.parametrize(
+    "field_targets",
+    [
+        "outlet1",
+        "outlet2",
+        ["outlet1", "outlet2"],
+        [],
+        ["outlet3"],
+        "outlet3",
+        (),
+        ("outlet1", "outlet2"),
+        ("outlet1", "outlet3"),
+    ],
+)
+def test_choice_by_field(field_name, field_targets):
+    choice_step = ChoiceByField(field_name=field_name)
+    fn = mlrun.new_function(
+        name="choice-by-field-example",
+        kind="serving",
+    )
+    graph = fn.set_topology("flow")
+    graph.to(name="generate_field", handler="generate_field").to(
+        choice_step, name="pick_outlets"
+    )
+    graph.add_step(
+        "storey.Extend",
+        name="outlet1",
+        _fn='({"field_visited": "outlet1"})',
+        after="pick_outlets",
+    )
+    graph.add_step(
+        "storey.Extend",
+        name="outlet2",
+        _fn='({"field_visited": "outlet2"})',
+        after="pick_outlets",
+    )
+    graph.add_step(
+        "storey.Extend", name="end", _fn="({})", after=["outlet1", "outlet2"]
+    ).respond()
+    fn_server = fn.to_mock_server()
+    try:
+        with pytest.raises(
+            (RuntimeError, mlrun.MLRunNotFoundError, ValueError)
+        ) as exc_info:
+            result = fn_server.test(
+                body={"field_name": field_name, "field_targets": field_targets}
+            )
+            assert isinstance(
+                exc_info.value, (RuntimeError, mlrun.MLRunNotFoundError, ValueError)
+            )
+    except AssertionError:
+        # means no exception was raised → handle as success case
+        if isinstance(field_targets, (list, tuple)):
+            if len(field_targets) < 2:
+                assert sorted(result["field_visited"]) == sorted(field_targets)
+            else:
+                assert result["field_visited"] in field_targets
+        else:
+            assert result["field_visited"] == field_targets
```

Comment:
The test doesn't terminate because the mock server isn't terminated. Missing a call to `fn_server.wait_for_completion()` at the end (in a finally block, or a fixture.)